File: java/AccessToken.java


public class AccessToken {


    public static void main(String[] args) {

        String accessKeyId = "accessKeyId";
        String accessKeySecret = "accessKeySecret";
        String mediaId = "mediaId";

        String createAccessToken = createAccessToken(accessKeyId, mediaId, accessKeySecret);

        System.out.println("createAccessToken: " + createAccessToken);
    }

    public static String createAccessToken(String accessKeyId, String mediaId, String accessKeySecret) {

        String ts = String.valueOf(System.currentTimeMillis());

        String tempStr = accessKeyId + "|" + ts + "|" + mediaId;

        String createAccessToken = XXTEA.encryptToBase64String(tempStr, accessKeySecret);
        createAccessToken = createAccessToken.replace("+", "-").replace("/", "_").replace("=", "*");

        return createAccessToken;
    }
}

File: go/AccessToken.go
package main

import (
    "fmt"
    "time"
    "strings"
    "strconv"
    "encoding/base64"
    "github.com/xxtea/xxtea-go/xxtea"
)

func createAccessToken(accessKeyId string, mediaId string, accessKeySecret string) string{

	ts := time.Now().UnixNano() / 1e6
    tsStr := strconv.FormatInt(ts,10)
    tempStr := accessKeyId + "|" + tsStr + "|" + mediaId
    createAccessToken := xxtea.Encrypt( []byte(tempStr), []byte(accessKeySecret))
	var createAccessTokenStr = base64.StdEncoding.EncodeToString(createAccessToken)
    createAccessTokenStr = strings.Replace(createAccessTokenStr,"+", "-",-1)
    createAccessTokenStr = strings.Replace(createAccessTokenStr,"/", "_",-1)
    createAccessTokenStr = strings.Replace(createAccessTokenStr,"=", "*",-1)

    return createAccessTokenStr
}

func main() {
    accessKeyId := "accessKeyId"
    mediaId := "mediaId"
    accessKeySecret := "accessKeySecret"

    fmt.Println("accessKeyId: ", accessKeyId)
		
    accessToken := createAccessToken(accessKeyId,mediaId,accessKeySecret)
    fmt.Println("accessToken: ", accessToken)
}
